File: Backend/Dixen.API/Dixen.Repo/Model/Entities/Evnt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Dixen.Repo.Model.Entities
{
    public class Evnt
    {
        public int Id { get; set; }
        public required string Title { get; set; } = string.Empty; 
        public required string Description { get; set; } = string.Empty;
        public DateTime StartTime { get; set; }
        public string ImageUrl { get; set; }
        public int OrganizerId { get; set; }
        public Organizer Organizer { get; set; }
        public List<Category> Categories { get; set; } = new();
        public List<Performer> Performers { get; set; }= new();
        //public List<EventAttendance> Attendances { get; set; }= new();
        public List<SocialShare> SocialShares { get; set; }= new();
        public List<EventSubmission> Submissions { get; set; }= new();

        // New property for Halls with individual capacities
        public List<Hall> Halls { get; set; }= new();
        public List<Booking> Bookings { get; set; }= new();
        public List<Ticket> Tickets { get; set; } = new();
        public List<EventReview> Reviews { get; set; } = new();

    }
}

File: Frontend/DixenUI/src/app/Models/Evnt.ts
import { Category } from './category';
import { Hall } from './hall';

export class Evnt {
  id: number = 0;
  title: string = '';
  description: string = '';
  startTime: Date = new Date();
  imageUrl: string = '';
  organizerId: number = 0;
  // categories!: Category[];
  categories?: Category[];
  categoryIds: number[] = [];
  halls?: Hall[];
  categoryNames?: string[];
  hallNames?: string[];
  organizerName: string = '';
  name?: string;
  address?: string;
  city?: string;
  tickets?: { id: number; type: string; price: number; quantity: number }[];
  minPrice?: number;
  maxPrice?: number;
  hallIds: number[] = [];
  date?: string;
}
